src/task/methods/jn_ap_op2/script.py

```diff
@@ -0,0 +1,474 @@
+    rng = np.random.default_rng(seed=seed)
+    
+    plant_seed(seed)
+    
+    # Initialize NN
+    n_input_channels = X.size()[2]
+    model = NN(X.shape[1], n_input_channels, **kwargs)
+    if USE_GPU:
+        model.cuda()
+        device='cuda'
+    else:
+        device=None
+    generator = torch.Generator(device=device).manual_seed(seed)
+    # Initialize optimizer and scheduler
+    optimizer = torch.optim.Adam(model.parameters(), lr=1e-3, eps=1e-7)
+    scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
+            optimizer, mode='min', factor=0.9, patience=5, verbose=False, threshold=0.0001,
+            threshold_mode='rel', cooldown=2, min_lr=1e-5, eps=1e-08)
+    
+    # Place data on GPU
+    if USE_GPU:
+        X = X.cuda()
+        Y = Y.cuda()
+    
+    pbar = tqdm.tqdm(range(n_iter))
+    for epoch in pbar:
+        total_loss, baseline_total_loss = 0, 0
+        rng.shuffle(idx_train)
+        for it_idx in np.array_split(idx_train, 200):  # 120
+            
+            # Reset gradients
+            optimizer.zero_grad()
+            
+            # Create batch
+            Y_target = Y[it_idx, :]
+            if USE_GPU:
+                Y_target = Y_target.cuda()
+            
+            # Sample from the null distribution and add background noise to the targets
+            Y_target = Y_target + 0.5 * background_noise(*Y_target.size(), device=X.device, generator=generator)
+            
+            # Forward pass
+            x = X[it_idx]
+            if USE_GPU:
+                x = x.cuda()
+            Y_pred = model.forward(x)
+            
+            # Compute loss (mean absolute error) in a differentiable fashion
+            loss = torch.mean(torch.abs(Y_target - Y_pred))
+            
+            # Backward pass
+            loss.backward()
+            
+            # Update NN parameters
+            optimizer.step()
+            
+            # Update training MRRMSE, and also update MRRMSE for the baseline (zero) predictor
+            # for comparison purposes
+            mrrmse = torch.sum(torch.sqrt(torch.mean(torch.square(Y_target - Y_pred), dim=1)))
+            baseline_mrrmse = torch.sum(torch.sqrt(torch.mean(torch.square(Y_target), dim=1)))
+            total_loss += mrrmse.item()
+            baseline_total_loss += baseline_mrrmse.item()
+            
+        # Compute relative error. Relative error is < 1 when the model performs better than
+        # the baseline on the training data.
+        rel_error = total_loss / baseline_total_loss
+        
+        # Update learning rate
+        scheduler.step(rel_error)
+        
+        # Show relative error
+        pbar.set_description(f'{rel_error:.3f}')
+    return model
+
+## VIASH START
+par = {
+  "de_train": "resources/neurips-2023-kaggle/de_train.parquet",
+  "de_test": "resources/neurips-2023-kaggle/de_test.parquet",
+  "id_map": "resources/neurips-2023-kaggle/id_map.csv",
+  "output": "output.parquet",
+}
+## VIASH END
```

Comment:
Once you've added an argument to the config, you can add it to `par` in the viash codeblock:

```python
## VIASH START
par = {
  "de_train": "resources/neurips-2023-kaggle/de_train.parquet",
  "de_test": "resources/neurips-2023-kaggle/de_test.parquet",
  "id_map": "resources/neurips-2023-kaggle/id_map.csv",
  "output": "output.parquet",
  "epochs": 10
}
## VIASH END
```

This will allow you to use `par["epochs"]` throughout your script.